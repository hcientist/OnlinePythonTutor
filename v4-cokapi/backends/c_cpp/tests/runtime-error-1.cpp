#include <stdlib.h>

int main() {
  int* x = (int*)malloc(2*sizeof(*x));
  x[100] = 10;   // invalid write
  int y = x[10]; // invalid read
}
