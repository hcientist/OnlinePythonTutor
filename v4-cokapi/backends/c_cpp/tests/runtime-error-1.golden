{
  "code": "#include <stdlib.h>\n\nint main() {\n  int* x = (int*)malloc(2*sizeof(*x));\n  x[100] = 10;   // invalid write\n  int y = x[10]; // invalid read\n}", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 3, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 3, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {}, 
      "line": 4, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {}, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 3, 
          "ordered_varnames": [], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {
        "0x53FE040": [
          "C_ARRAY", 
          "0x53FE040", 
          [
            "C_DATA", 
            "0x53FE040", 
            "int", 
            "<UNINITIALIZED>"
          ], 
          [
            "C_DATA", 
            "0x53FE044", 
            "int", 
            "<UNINITIALIZED>"
          ]
        ]
      }, 
      "line": 5, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "0x53FE040"
            ], 
            "y": [
              "C_DATA", 
              "0xFFF000104", 
              "int", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 5, 
          "ordered_varnames": [
            "x", 
            "y"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "exception", 
      "exception_msg": "ERROR: Invalid write of size 4\n(Stopped running after the first error. Please fix your code.)", 
      "func_name": "main", 
      "globals": {}, 
      "heap": {
        "0x53FE040": [
          "C_ARRAY", 
          "0x53FE040", 
          [
            "C_DATA", 
            "0x53FE040", 
            "int", 
            "<UNINITIALIZED>"
          ], 
          [
            "C_DATA", 
            "0x53FE044", 
            "int", 
            "<UNINITIALIZED>"
          ]
        ]
      }, 
      "line": 5, 
      "ordered_globals": [], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x": [
              "C_DATA", 
              "0xFFF000108", 
              "pointer", 
              "0x53FE040"
            ], 
            "y": [
              "C_DATA", 
              "0xFFF000104", 
              "int", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 5, 
          "ordered_varnames": [
            "x", 
            "y"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }
  ]
}

